plugins/gnupgplugin/gpgmebase.cpp

```diff
@@ -0,0 +1,761 @@
+                                             const Key &key,
+                                             CryptoManager::BlockMode blockMode,
+                                             CryptoManager::EncryptionPadding padding,
+                                             const QByteArray &authenticationData,
+                                             const QVariantMap &customParameters,
+                                             QByteArray *encrypted,
+                                             QByteArray *authenticationTag)
+{
+    Q_UNUSED(iv);
+    Q_UNUSED(blockMode);
+    Q_UNUSED(padding);
+    Q_UNUSED(authenticationData);
+    Q_UNUSED(authenticationTag);
+
+    return operation(CryptoManager::OperationEncrypt,
+                     key, data, customParameters, encrypted);
+}
+
+Result Daemon::Plugins::GnuPGPlugin::decrypt(const QByteArray &data,
+                                             const QByteArray &iv,
+                                             const Key &key, // or keyreference, i.e. Key(keyName)
+                                             CryptoManager::BlockMode blockMode,
+                                             CryptoManager::EncryptionPadding padding,
+                                             const QByteArray &authenticationData,
+                                             const QByteArray &authenticationTag,
+                                             const QVariantMap &customParameters,
+                                             QByteArray *decrypted,
+                                             CryptoManager::VerificationStatus *verificationStatus)
+{
+    Q_UNUSED(iv);
+    Q_UNUSED(blockMode);
+    Q_UNUSED(padding);
+    Q_UNUSED(authenticationData);
+    Q_UNUSED(authenticationTag);
+    Q_UNUSED(customParameters);
+
+    if (!verificationStatus) {
+        return Result(Result::CryptoPluginDecryptionError,
+                      QStringLiteral("missing verificationStatus argument."));
+    }
+    *verificationStatus = CryptoManager::VerificationStatusUnknown;
+
+    if (key.storagePluginName() != name()) {
+        return Result(Result::CryptoPluginDecryptionError,
+                      QStringLiteral("cannot decrypt with a non GnuPG key."));
+    }
+
+    GPGmeData gdata(data), output;
+    if (!gdata || !output) {
+        return Result(Result::CryptoPluginDecryptionError,
+                      QStringLiteral("cannot create cryptographic data."));
+    }
+
+    GPGmeContext ctx(m_protocol, key.filterData("Ephemeral-Home"));
+    if (!ctx)
+        return Result(Result::CryptoPluginDecryptionError, ctx.error());
+
+    gpgme_error_t err;
+    err = gpgme_op_decrypt_verify(ctx, gdata, output);
+    if (gpgme_err_code(err) != GPG_ERR_NO_ERROR) {
+        return Result(Result::CryptoPluginDecryptionError,
+                      QStringLiteral("cannot decrypt message: %1.").arg(gpgme_strerror(err)));
+    }
+    Result verifyResult = verify(CryptoManager::OperationDecrypt,
```

Comment:
What does this do?

Reply:
Name of function changed for checkGPGmeVerification() which is hopefully more descriptive.